### PR TITLE
Fix the implementation of the HLG EOTF function

### PR DESCRIPTION
**Pull request by krasuluk**
This PR fixes the implementation of HLG EOTF function. An important note is also added in case the function was to be used on channels other than Y.

### CODE REVIEW

File: siti_tools/siti.py [hunk 1]
```diff
@@ -482,19 +482,24 @@ def eotf_hlg(
         Returns:
             frame_data: pixel values in the physical luminance up to 10,000 cd/m^2
         """
-        frame_data = np.maximum(frame_data, 0.0)
```
Comment (krasuluk) [inline siti_tools/siti.py:485]: According to the [ITU-R Recommendation BT.2100](https://www.itu.int/dms_pubrec/itu-r/rec/bt/R-REC-BT.2100-1-201706-S!!PDF-E.pdf), the values below 0 and above 1 are also possible and acceptable (see NOTE 5g).

File: siti_tools/siti.py [hunk 2]
```diff
@@ -482,19 +482,24 @@ def eotf_hlg(
...
-        frame_data = np.minimum(frame_data, 1.0)
 
         a = 0.17883277
         b = 0.02372241
         c = 1.00429347
-        gamma = 1.2
+        if l_max <= 1000.0:
+            gamma = 1.2
+        else:
+            gamma = 1.2 + 0.42 * np.log10(l_max/1000.0)
```
Comment (krasuluk) [inline siti_tools/siti.py:492]: See NOTE 5e in the Recommendation BT.2100

File: siti_tools/siti.py [hunk 3]
```diff
@@ -482,19 +482,24 @@ def eotf_hlg(
...
         frame_data = (frame_data <= 0.5) * (np.power(frame_data, 2.0) / 3.0) + (
             frame_data > 0.5
         ) * (np.exp((frame_data - c) / a) - b)
 
-        frame_data = (l_max - l_min) * np.power(frame_data, gamma - 1.0) + l_min
+        # NOTE: The following is only valid when the function is applied to the Y channel.
+        # If frame_data was a color channel (R, G, or B), the correct implementation would be:
+        # (l_max - l_min) * np.power(Y, gamma - 1.0) * frame_data + l_min,
+        # where Y would be obtained from linearized R, G, B channels.
+        frame_data = (l_max - l_min) * np.power(frame_data, gamma) + l_min
```
Comment (krasuluk) [inline siti_tools/siti.py:502]: If `frame_data = Y` the equation in the NOTE can be written as:

`(l_max - l_min) * np.power(frame_data, gamma - 1.0 + 1.0) + l_min`

**Comment by krasuluk** [comment]
@slhck the fix results in test failures, as expected. I assume those were obtained from the original implementation which I believe had a bug (brought to my attention by @cosmin). Can you double-check the proposed changes make sense? Once verified, we can fix the failing tests.

**Review by slhck — APPROVED** [review]

File: siti_tools/siti.py (hunk 3 above)
Comment (slhck) [reply siti_tools/siti.py:502]: This part I don't see in BT.2100, but to be honest I have only looked very briefly.

**Comment by slhck** [comment]
Feel free to merge and fix the tests after, or change the values before! Thanks to @cosmin and you for providing the fix!

File: siti_tools/siti.py (hunk 3 above)
Comment (krasuluk) [reply siti_tools/siti.py:502]: @slhck You are right, it is not in the recommendation as it only defines the functions for the individual components. The "most correct" way would therefore be to convert the YUV input to RGB, apply the EOTF on individual components, and convert back to YUV. But I believe that is not really needed for our use case and this solution seems like a reasonable approximation. Would you agree?

Comment (slhck) [reply siti_tools/siti.py:502]: I see, yes, that makes sense. I was more confused about the subtraction `gamma - 1.0` in the original code and where that came from.

**Comment by slhck** [comment]
Will do a release tomorrow, thanks!

**Pull request merged**